### PR TITLE
Add experts documentation

### PR DESCRIPTION
**Pull request by ernestum**
## Description

This PR documents how experts can be loaded from disk and from HuggingFace. It also describes how to push your own expert policies to HuggingFace.

Note: this PR branchess off #565 and should be rebased on master after #565 as been merged.

### CODE REVIEW

File: docs/experts/loading-experts.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+By default, the policies are loaded from the HumanCompatibleAI repository, but you can
+override this by setting the `organization` parameter in the `loader_kwargs`.
+When using the python API, you also have to speciyf the environment name as `env_name`.
+
+Uploading a policy to HuggingFace
+---------------------------------
+
+The `huggingface-sb3 package <https://github.com/huggingface/huggingface_sb3>`_ provides
+utilities to push your own models to HuggingFace and load them from there.
+Make sure use the naming scheme helpers
```
Comment (dfilan) [inline docs/experts/loading-experts.rst:44]: Should be "Make sure to use"

File: docs/experts/loading-experts.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Loading a policy from HuggingFace
+---------------------------------
+
+`HuggingFace <https://huggingface.co/>`_ is a popular repository for pretrained models.
+
+To load a policy from HuggingFace, use either `ppo-huggingface` or `sac-huggingface` as
+the policy type.
+By default, the policies are loaded from the HumanCompatibleAI repository, but you can
+override this by setting the `organization` parameter in the `loader_kwargs`.
+When using the python API, you also have to speciyf the environment name as `env_name`.
```
Comment (dfilan) [inline docs/experts/loading-experts.rst:37]: should be specify

Comment (ernestum) [reply docs/experts/loading-experts.rst]: thanks, fixed

File: docs/experts/loading-experts.rst (hunk 1 above)
Comment (ernestum) [reply docs/experts/loading-experts.rst]: thanks, fixed

File: docs/experts/loading-experts.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
+===============
+Loading Experts
+===============
+
+Some algorithms, such as DAgger, can learn from an expert policy.
+
+In the :doc:`../getting-started/first-steps` tutorial, we first trained an expert and
+then imitate it's behavior using :doc:`../algorithms/bc`.
+In practice, you want to load a pre-trained policy for performance reasons.
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:9]: "you want" is a bit strong -- we don't know the user's situation. Could also soften it to "you typicaly want" etc.
```suggestion
In practice, you may want to load a pre-trained policy for performance reasons.
```

File: docs/experts/loading-experts.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+The python interface provides a :func:`~imitation.policies.serialize.load_policy`
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:11]:
```suggestion
The Python interface provides a :func:`~imitation.policies.serialize.load_policy`
```

File: docs/experts/loading-experts.rst [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+function to which you pass a `policy type` and any extra kwargs to pass to the
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:12]:
```suggestion
function to which you pass a `policy_type` and any extra kwargs to pass to the
```

File: docs/experts/loading-experts.rst [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+`expert.loader_kwargs` parameters.
+
+There are several existing policy types, and you can also define your own in the
+registry.
+
+Loading a policy from a file
+----------------------------
+
+To load a policy from disk, use either `ppo` or `sac` as the policy type.
+The path is specified as `path` to the `loader_kwargs` and it should either point
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:25]:
```suggestion
The path is specified by `path` in the `loader_kwargs` and it should either point
```

File: docs/experts/loading-experts.rst [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+Loading a policy from HuggingFace
+---------------------------------
+
+`HuggingFace <https://huggingface.co/>`_ is a popular repository for pre-trained models.
+
+To load a policy from HuggingFace, use either `ppo-huggingface` or `sac-huggingface` as
+the policy type.
+By default, the policies are loaded from the HumanCompatibleAI repository, but you can
+override this by setting the `organization` parameter in the `loader_kwargs`.
+When using the python API, you also have to specify the environment name as `env_name`.
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:37]:
```suggestion
When using the Python API, you also have to specify the environment name as `env_name`.
```

**Review by AdamGleave — COMMENTED** [review]
The documentation as written looks good apart from a few minor comments, but it seems to be missing a bit. It only describes how to use it to brain BC/DAgger with experts, but can't we also sample demonstrations from pre-trained experts with GAIL/AIRL (and perhaps density and MCE)? The latter use case seems more common so it'd be great to have documentation for that too, although if you wanted to split that into a separate PR I wouldn't object.

**Comment by ernestum** [comment]
We don't have scripts for density and MCE and they need demonstrations (not an expert).
For GAIL/AIRL, if no demonstrations are specified, they are sampled from an expert ad-hoc. I will mention this and add some examples.

**Comment by ernestum** [comment]
I added a proper introduction, addressed the case with AIRL/GAIL and added a section on how to register custom experts.
Copilot was surprisingly good at generating usage examples so we have them too.
The new `config_hook` in `expert.py` makes the `loader_kwargs` API somewhat discoverable without digging into the documentation.

File: docs/experts/loading-experts.rst [hunk 8]
```diff
@@ -0,0 +1,125 @@
+===============
+Loading Experts
+===============
+
+The algorithms in the imitation library are all about learning from some kind of
+expert.
+In many cases it is practical that this expert is a piece of software itself.
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:7]: "practical that" sounds a bit off to my ears, possible rephrasing below, there might be others that better capture your intention.
```suggestion
In many cases it is simplest if this expert is a piece of software itself.
```

File: docs/experts/loading-experts.rst [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+The library natively supports experts trained using the
+`stable-baselines3 <https://github.com/DLR-RM/stable-baselines3>`_ reinforcement
+learning library.
+
+For example, BC and DAgger can learn from an expert policy and the command line
+interface to AIRL/GAIL allows to specify an expert to sample demonstrations from.
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:13]:
```suggestion
interface of AIRL/GAIL allows one to specify an expert to sample demonstrations from.
```

File: docs/experts/loading-experts.rst [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+    venv = util.make_vec_env("your-env", n_envs=4)
+    local_policy = load_policy("ppo", venv, loader_kwargs={"path": "path/to/model.zip"})
+
+To load a policy from disk, use either `ppo` or `sac` as the policy type.
+The path is specified by `path` in the `loader_kwargs` and it should either point
+to a zip file containing the policy or a directory containing a `model.zip` file that
+was created by stable-baselines3.
+
+In command line interface the `expert.policy_type` and `expert.loader_kwargs`
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:40]:
```suggestion
In the command line interface the `expert.policy_type` and `expert.loader_kwargs`
```

File: docs/experts/loading-experts.rst [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+Make sure to use the naming scheme helpers
+`as described in the readme <https://github.com/huggingface/huggingface_sb3#case-5-i-want-to-automate-uploaddownload-from-the-hub>`_.
+Otherwise, the loader will not be able to find your model in the repository.
+
+For a convenient high-level interface to train RL models and upload them to HuggingFace,
+we recommend using the
+`rl-baselines3-zoo <https://github.com/DLR-RM/rl-baselines3-zoo/>`_.
+
+
+Custom expert types
```
Comment (AdamGleave) [inline docs/experts/loading-experts.rst:100]: Nice, hadn't thought to document this but it's an important part of the API :)

File: src/imitation/scripts/common/expert.py [hunk 12]
```diff
@@ -9,20 +9,37 @@
...
     # See imitation.policies.serialize.load_policy for options.
     loader_kwargs = dict()
     locals()  # quieten flake8
 
 
-@expert_ingredient.capture
-def get_expert_policy(venv, policy_type, loader_kwargs, common):
-    if "huggingface" in policy_type:
+@expert_ingredient.config_hook
+def config_hook(config, command_name, logger):
```
Comment (AdamGleave) [inline src/imitation/scripts/common/expert.py:20]: Nice, this definitely makes things more self-documenting :)

**Review by AdamGleave — APPROVED** [review]
LGTM, this is much more comprehensive! I've added some minor suggestions re: phrasing. Feel free to merge after addressing those.

**Pull request merged**